web/source/osk/mouseEventEngine.ts

```diff
@@ -0,0 +1,81 @@
+/// <reference path="inputEventEngine.ts" />
+
+namespace com.keyman.osk {
+  export class MouseEventEngine extends InputEventEngine {
+    private readonly _mouseStart: typeof MouseEventEngine.prototype.onMouseStart;
+    private readonly _mouseMove:  typeof MouseEventEngine.prototype.onMouseMove;
+    private readonly _mouseEnd:   typeof MouseEventEngine.prototype.onMouseEnd;
+
+    private hasActiveClick: boolean = false;
+
+    public constructor(vkbd: VisualKeyboard) {
+      super(vkbd);
+
+      this._mouseStart = this.onMouseStart.bind(this);
+      this._mouseMove  = this.onMouseMove.bind(this);
+      this._mouseEnd   = this.onMouseEnd.bind(this);
+    }
+
+    registerEventHandlers() {
```

Comment:
I wonder if these functions should be idempotent?

Reply:
In what way?  Unbind first, just in case, before binding?

I've written them this way to ensure that we can easily "turn off" the event handlers for the two different OSK input types easily and independently.  This is most obviously useful during a deconstructor-like operation, but it should also be useful during development testing - both for KeymanWeb and for keyboards as part of a test page.

Reply:
That was my thinking but now I see it may not be necessary, now that I've found the appropriate ref: https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/addEventListener#multiple_identical_event_listeners